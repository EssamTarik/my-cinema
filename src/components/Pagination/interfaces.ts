export interface IProps {
  page: number;
  totalPages: number;
  onNextClick: () => void;
  onPreviousClick: () => void;
}