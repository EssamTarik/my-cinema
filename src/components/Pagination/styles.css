.pagination {
  display: flex;
  padding-bottom: 15px;
}
.pagination > * {
  flex: 1;
}

.pagination__next-container {
  text-align: right;
  padding-right: 15px;
}
.pagination__prev-container {
  text-align: left;
  padding-left: 15px;
}
.pagination__btn {
  padding: 2px 10px;
  outline: none;
  border: 1px solid lightgray;
  border-radius: 4px;
  cursor: pointer;

  background-color: #0A1016;
  color: white;
}
.pagination__btn:disabled {
  color: grey;
  cursor: auto;
}