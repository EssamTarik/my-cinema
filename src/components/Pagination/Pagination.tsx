import React from "react";
import cn from "classnames";
import { FormattedMessage } from "react-intl";
import { IProps } from "./interfaces";
import "./styles.css";

const Pagination = ({
  page,
  totalPages,
  onNextClick,
  onPreviousClick,
}: IProps) => {
  const isLastPage = page === totalPages;
  const isFirstPage = page === 1;

  return (
    <div className="pagination">
      <div className="pagination__prev-container">
        <button
          type="button"
          onClick={onPreviousClick}
          disabled={isFirstPage}
          className={cn("pagination__btn")}
        >
          <FormattedMessage id="pagination.previous" />
        </button>
      </div>
      <div className="pagination__next-container">
        <button
          type="button"
          onClick={onNextClick}
          disabled={isLastPage}
          className={cn("pagination__btn")}
        >
          <FormattedMessage id="pagination.next" />
        </button>
      </div>
    </div>
  );
};

export default Pagination;
