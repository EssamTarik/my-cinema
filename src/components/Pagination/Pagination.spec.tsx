jest.mock('react-intl', () => ({ FormattedMessage: () => <div /> }));
import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";

import Pagination from "./Pagination";

describe('Pagination', () => {
  afterEach(cleanup);
  it("renders", () => {
    const { asFragment } = render(
      <Pagination
        page={1}
        totalPages={2}
        onNextClick={jest.fn()}
        onPreviousClick={jest.fn()}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('disables next button on last page', () => {
    const { container } = render(
      <Pagination
        page={10}
        totalPages={10}
        onNextClick={jest.fn()}
        onPreviousClick={jest.fn()}
      />
    );
    const nextBtn = container.querySelector('.pagination__next-container button');
    expect(nextBtn!).toBeDisabled()
  })

  it('disables previous button on first page', () => {
    const { container } = render(
      <Pagination
        page={1}
        totalPages={10}
        onNextClick={jest.fn()}
        onPreviousClick={jest.fn()}
      />
    );
    const previousBtn = container.querySelector('.pagination__prev-container button');
    expect(previousBtn!).toBeDisabled()
  })

  it('calls next/previous page callback', () => {
    const mockNextClick = jest.fn()
    const mockPreviousClick = jest.fn()
    const { container } = render(
      <Pagination
        page={2}
        totalPages={10}
        onNextClick={mockNextClick}
        onPreviousClick={mockPreviousClick}
      />
    );

    const previousBtn = container.querySelector('.pagination__prev-container button');
    const nextBtn = container.querySelector('.pagination__next-container button');
    fireEvent.click(previousBtn!);
    fireEvent.click(nextBtn!);

    expect(mockNextClick).toHaveBeenCalledTimes(1);
    expect(mockPreviousClick).toHaveBeenCalledTimes(1);
  })
})

