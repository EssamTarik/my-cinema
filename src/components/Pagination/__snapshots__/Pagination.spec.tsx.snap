// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders 1`] = `
<DocumentFragment>
  <div
    class="pagination"
  >
    <div
      class="pagination__prev-container"
    >
      <button
        class="pagination__btn"
        disabled=""
        type="button"
      >
        <div />
      </button>
    </div>
    <div
      class="pagination__next-container"
    >
      <button
        class="pagination__btn"
        type="button"
      >
        <div />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
